packages/sources/finage/src/endpoint/etf.ts

```diff
@@ -0,0 +1,55 @@
+import type { Config, ExecuteWithConfig, InputParameters } from '@chainlink/ea-bootstrap'
+import { Requester, util, Validator } from '@chainlink/ea-bootstrap'
+import { NAME } from '../config'
+import overrides from '../config/symbols.json'
+
+export const supportedEndpoints = ['etf']
+export const batchablePropertyPath = [{ name: 'base' }]
+
+export const description = `https://finage.co.uk/docs/api/etf-last-price
+The result will be the price field in response.`
+
+export type TInputParameters = { base: string }
+export const inputParameters: InputParameters<TInputParameters> = {
+  base: {
+    required: true,
+    aliases: ['from', 'symbol'],
+    description: 'The symbol of the etf to query',
+  },
+}
+
+export interface ResponseSchema {
+  symbol: string
+  price: number
+  timestamp: number
+}
+
+export const makeEtfExecute =
+  (executeConfig: { country?: string }): ExecuteWithConfig<Config> =>
+  async (request, _, config) => {
+    const validator = new Validator(request, inputParameters, {}, { overrides })
+
+    const jobRunID = validator.validated.id
+    const base = validator.validated.data.base
+    const symbol = validator.overrideSymbol(NAME, base).toUpperCase()
+
+    const url = util.buildUrlPath('/last/etf/:symbol', { symbol })
+
+    const params = {
+      apikey: config.apiKey,
+      country: executeConfig.country,
```

Comment:
is there any risk that `undefined` ends up in the url params at some point?

Reply:
It shouldn't, as these params are passed to Axios which filters out undefined params.

However to be safe incase that changed in the future, I've pushed a change so that we only set `country` if `executeConfig.country` is defined.